### PR TITLE
test: ecdsa stake registry unit tests

### PR DESCRIPTION
**Pull request by stevennevins**
**Motivation:**

Improve coverage on ECDSA Stake Registry 

**Modifications:**

Add tests for edge cases for reverts with Quorum Config updates and tests for event emission

**Result:**

improved test coverage

### CODE REVIEW

File: test/mocks/ECDSAStakeRegistryMock.sol [hunk 1]
```diff
@@ -2,13 +2,49 @@
...
         IDelegationManager _delegationManager
     ) ECDSAStakeRegistry(_delegationManager) {}
+
+    /**
+     * @notice Sets the operator weight at a specific block for testing
+     * @param operator The operator address
+     * @param blockNumber The block number
+     * @param weight The weight to set
+     */
+    function setOperatorWeightAtBlock(
```
Comment (Sidu28) [inline test/mocks/ECDSAStakeRegistryMock.sol]: I'm not seeing these functions being used?

Comment (stevennevins) [reply test/mocks/ECDSAStakeRegistryMock.sol]: Oh yeah good call. I think I intended to use it in a test but stopped

**Review by Sidu28 — APPROVED** [review]
LGTM otherwise

File: test/mocks/ECDSAStakeRegistryMock.sol (hunk 1 above)
Comment (stevennevins) [reply test/mocks/ECDSAStakeRegistryMock.sol]: Removed: [e988f4c](https://github.com/Layr-Labs/eigenlayer-middleware/pull/436/commits/e988f4c02a5a66f9922f409dd4fa08954fa1adb0)
